src/rgw/rgw_admin.cc

```diff
@@ -10557,6 +10562,30 @@ int main(int argc, const char **argv)
     }
   }
 
+  if (opt_cmd == OPT::PUBSUB_TOPIC_STATS  ) {
+    if (topic_name.empty()) {
+      cerr << "ERROR: topic name was not provided (via --topic)" << std::endl;
+      return EINVAL;
+    }
+
+    std::size_t number_of_reservations;
+    uint32_t number_of_entries;
+    uint64_t queue_size;
+    ret = rgw::notify::get_persistent_queue_by_topic_name(
+        dpp(), static_cast<rgw::sal::RadosStore*>(driver)->getRados()->get_notif_pool_ctx(), topic_name, number_of_reservations,
+        number_of_entries, queue_size, null_yield);
+    if (ret == -ENOENT) {
+      // queue list object was not created - nothing to do
+      std::cout << "persistent queue was not created" << std::endl;
+      return 0;
+    } else if (ret < 0) {
+      cerr << "ERROR: could not get persistent queue: " << cpp_strerror(-ret) << std::endl;
+      return -ret;
+    }
+    std::cout << "persistent queue size: " << queue_size << "(bytes), number of entries: " << number_of_entries
```

Comment:
coudl you format the output in JSON?
see `bucket stats` as an example.

Reply:
Ack!